<launch>
    <arg name="Joffsets" default="[40, 40, 5, 60, 80, 80]"/>

    <node name="follow_key_input" pkg="sitl_dvrk" type="follow_key_input.py" output="screen">
    </node>

    <node name="follow_tissue_bnd" pkg="sitl_dvrk" type="follow_tissue_bnd.py" output="screen">
        <param name="arm_name"           type="str"       value="PSM1"/>
        <param name="expected_interval"  type="double"    value="0.01"/>
        <param name="bnd_msg"            type="str"       value="/seg/liver_gallb/bnd_3d"/>
        <param name="tf_path"            type="str"       value="/home/$(env USER)/aruco_data/base_tfs.yaml"/>
        <param name="calib_fn"           type="str"       value="/home/$(env USER)/aruco_data/psm1_calib_results_final_new_v2.mat"/>
        <param name="num_success"        type="int"       value="8"/>
        <param name="move_dist"          type="double"    value="5e-4"/>
        <param name="d_eps"              type="double"    value="1e-3"/>
        <param name="pedal_flag"         type="bool"      value="true"/>
        <param name="Tweight"            type="double"    value="2.0"/>
        <param name="Rweight"            type="double"    value="0.2"/>
        <param name="window_size"        type="int"       value="15"/>
        <param name="window_thr"         type="double"    value="3e-4"/>
        <rosparam param="Joffsets" subst_value="True"> $(arg Joffsets) </rosparam>
    </node>

    <?ignore

    ?>
</launch>